widget/icon.go

```diff
@@ -54,12 +57,14 @@ func (i *iconRenderer) updateObjects() {
 type Icon struct {
 	BaseWidget
 
-	Resource fyne.Resource // The resource for this icon
+	Resource  fyne.Resource // The resource for this icon
+	cachedRes fyne.Resource
 }
 
 // SetResource updates the resource rendered in this icon widget
 func (i *Icon) SetResource(res fyne.Resource) {
 	i.Resource = res
+	i.cachedRes = res
```

Comment:
This looks wrong - doesn't it mean that changing the resource will not bother resetting objects because the cache will be deemed the same as the resource...? Surely setting it to nil would be right?

Reply:
Good catch. Fixed that :)